components/central-application-gateway/pkg/authorization/oauth/oauthclient.go

```diff
@@ -58,6 +60,22 @@ func (c *client) GetToken(clientID, clientSecret, authURL string, headers, query
 	return tokenResponse.AccessToken, nil
 }
 
+func (c *client) GetTokenMTLS(clientID, authURL string, cert tls.Certificate, headers, queryParameters *map[string][]string) (string, apperrors.AppError) {
+	token, found := c.tokenCache.Get(clientID)
```

Comment:
Question: is it safe to reuse the same cache for both OAuth methods? Is it possible that there will be two APIs that are registered with the same `clientID` but different type (one is OAuth grant flow and the second is OAuth with mTLS)?